File: misc/js/random_alphanumeric_string_generator.js
function generateRandomAlphaNumericString(len = 13){
    const chars = []

    // uppercase-alphabets: A-Z
    for(let i=0; i<26; i++){
        chars.push(String.fromCharCode(65 + i))
    }

    // digits: 0-9
    for(let i=0; i<=9; i++){
        chars.push(String.fromCharCode(48 + i))
    }

    // lowercase-alphabets: a-z
    for(let i=0; i<26; i++){
        chars.push(String.fromCharCode(97 + i))
    }

    let randStr = ''
    while(len-- > 0){
        const randIdx = Math.floor(Math.random() * chars.length)
        randStr += chars[randIdx]
    }

    return randStr
}

// number of strings to be generated
let count = 100
// length of each string (default is 13)
const len = 13

console.log(`Generating ${count} alpha-numeric strings of length ${len} ...\n`)

for(let i=1; i<=count; i++){
    console.log(i + ' -> ' + generateRandomAlphaNumericString(len))
}

console.log('\n--------------------------------')

File: misc/cpp/random_alphanumeric_string_generator.cpp
#include <bits/stdc++.h>
using namespace std;

string generateRandomAlphaNumericString(int len){
    vector<char> chars;
    
    // uppercase-alphabets: A-Z
    for(char ch = 'A'; ch <= 'Z'; ch++)
        chars.push_back(ch);
    
    // digits: 0-9
    for(char ch = '0'; ch <= '9'; ch++)
        chars.push_back(ch);
    
    // lowercase-alphabets: a-z
    for(char ch = 'a'; ch <= 'z'; ch++)
        chars.push_back(ch);
    
    string randStr = "";
    
    while(len-- > 0)
    {
        int randIdx = rand() % chars.size();
        randStr += chars[randIdx];
    }
    
    return randStr;
}

int main(){
    // number of strings to be generated
    int count = 100;
    // length of each string
    int len = 13;
    
    srand(time(0)); // seed for rand()
    
    cout<<"Generating "<<count<<" random alphanumeric strings of length = "<<len<<" ... "<<endl<<endl;
    
    for(int i=1; i<=count; i++)
        cout<<i<<" -> "<<generateRandomAlphaNumericString(len)<<endl;
    
    cout<<endl<<string(25, '-')<<endl;
    
    return 0;
}